rust_snuba/src/processors/replays.rs

```diff
@@ -11,145 +11,166 @@ pub fn process_message(
     metadata: KafkaMessageMetadata,
 ) -> anyhow::Result<InsertBatch> {
     let payload_bytes = payload.payload().context("Expected payload")?;
+    let replay_row = deserialize_message(payload_bytes, metadata.partition, metadata.offset)?;
+    InsertBatch::from_rows([replay_row])
+}
 
-    let replay_message: ReplayMessage = serde_json::from_slice(payload_bytes)?;
+pub fn deserialize_message(
+    payload: &[u8],
+    partition: u16,
+    offset: u64,
+) -> Result<Vec<ReplayRow>, serde_json::Error> {
+    let replay_message: ReplayMessage = serde_json::from_slice(payload)?;
     let replay_payload = serde_json::from_slice(&replay_message.payload)?;
 
     match replay_payload {
-        ReplayPayload::ClickEvent(event) => {
-            InsertBatch::from_rows(event.clicks.into_iter().map(|click| ReplayRow {
+        ReplayPayload::ClickEvent(event) => Ok(event
+            .clicks
+            .into_iter()
+            .map(|click| ReplayRow {
                 click_alt: click.alt,
                 click_aria_label: click.aria_label,
                 click_class: click.class,
+                click_component_name: click.component_name,
                 click_id: click.id,
                 click_is_dead: click.is_dead,
                 click_is_rage: click.is_rage,
                 click_node_id: click.node_id,
-                click_component_name: click.component_name,
                 click_role: click.role,
                 click_tag: click.tag,
                 click_testid: click.testid,
                 click_text: click.text,
                 click_title: click.title,
+                error_sample_rate: -1.0,
                 event_hash: click.event_hash,
-                offset: metadata.offset,
-                partition: metadata.partition,
+                offset,
+                partition,
+                platform: "javascript".to_string(),
                 project_id: replay_message.project_id,
                 replay_id: replay_message.replay_id,
                 retention_days: replay_message.retention_days,
+                session_sample_rate: -1.0,
                 timestamp: click.timestamp as u32,
                 ..Default::default()
-            }))
-        }
+            })
+            .collect()),
         ReplayPayload::Event(event) => {
             let event_hash = match (event.event_hash, event.segment_id) {
                 (None, None) => Uuid::new_v4(),
                 (None, Some(segment_id)) => {
-                    Uuid::from_slice(md5::compute(segment_id.to_string().as_bytes()).as_slice())?
+                    Uuid::from_slice(md5::compute(segment_id.to_string().as_bytes()).as_slice())
+                        .expect("md5 hash never exceeds 16 bytes")
```

Comment:
Is it cool to `expect` here if we know its impossible to trigger?

Reply:
that should be fine, yes. Though you are only doing this because of the error type I believe?

Reply:
Yeah I didn't want to map the error to a serde type.  Anyhow didn't like `anyhow::Result<<Vec<ReplayRow>>>` for some reason.  It's saying "expected :: not >".  Which I'm not exactly sure where it wants the double colon but syntactically I didn't understand how to resolve in Rust.

Reply:
@cmanallen it's a syntax error, you have `<<` and `>>`, that doesn't work, need to replace with `<` and `>`

Reply:
It's still early lol.  Thank you.  Last night I was doing this `anyhow::Vec<ReplayRow>` which was failing because I didn't have the `Result` wrapping.  Which I didn't realize until 2 minutes ago when I left that comment and I still made an error... good grief.